csharp/src/Ice/Reference.cs

```diff
@@ -1330,12 +1248,10 @@ internal async ValueTask<IRequestHandler> GetConnectionRequestHandlerAsync(Cance
                 bool compress = false;
                 if (Protocol == Protocol.Ice1)
                 {
-                    // The connection endpoint is always a non-compressed endpoint. If one of the resolved endpoints
-                    // matches the connection endpoint, we don't use compression. Otherwise, if we don't find the
-                    // matching endpoint, it implies that we got a connection for one the compressed resolved endpoints.
-                    compress = !endpoints.Select(endpoint => connection.Endpoint == endpoint).Any();
-
-                    Debug.Assert(!compress || endpoints.Contains(connection.Endpoint.NewCompressionFlag(true)));
+                    // If one of the resolved proxy's endpoints matches one of the connection endpoint and this endpoint
```

Comment:
I don't find the check that complicated, if we can find an endpoint without the compress flag set, we prefer to use it otherwise we use compression. Anyway, I'll change it.